### PR TITLE
Add share component

### PR DESCRIPTION
**Pull request by gabriel-glo**
Share component lists 4 networks that support share links.

On mobile devices should open a native share dialog.

I didn't figure out how to control the display of certain social networks in  component options, so feedback is welcome on that side.

### CODE REVIEW

File: blocks/init/src/Blocks/components/share/share.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+$selectorClass = $attributes['selectorClass'] ?? $componentClass;
+$blockClass = $attributes['blockClass'] ?? '';
+
+$shareClass = Components::classnames([
+	$componentClass,
+	Components::selector($blockClass, $blockClass, "{$selectorClass}"),
+]);
+
+$shareItemClass = Components::classnames([
+	Components::selector($componentClass, $componentClass, 'item',),
```
Comment (iruzevic) [inline blocks/init/src/Blocks/components/share/share.php:34]: you have a rouge comma here

File: blocks/init/src/Blocks/components/share/share.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	Components::selector($blockClass, $blockClass, 'item',),
```
Comment (iruzevic) [inline blocks/init/src/Blocks/components/share/share.php:35]: you have a rouge comma here

File: blocks/init/src/Blocks/components/share/share.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	'pinterest' => "https://pinterest.com/pin/create/button/?url={$postUrl}&media={$postImage}&description={$postTitle}",
+];
+
+$componentClass = $attributes['componentClass'] ?? $manifest['componentClass'];
+$selectorClass = $attributes['selectorClass'] ?? $componentClass;
+$blockClass = $attributes['blockClass'] ?? '';
+
+$shareClass = Components::classnames([
+	$componentClass,
+	Components::selector($blockClass, $blockClass, "{$selectorClass}"),
```
Comment (iruzevic) [inline blocks/init/src/Blocks/components/share/share.php:30]: no need for template string here

File: blocks/init/src/Blocks/components/share/share.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * Template for the Share Component view.
+ *
+ * @package EightshiftBoilerplate
+ */
+
+use EightshiftBoilerplateVendor\EightshiftLibs\Helpers\Components;
+
+$manifest = Components::getManifest(__DIR__);
+
+$postUrl = \get_the_permalink();
```
Comment (iruzevic) [inline blocks/init/src/Blocks/components/share/share.php:13]: this options should be passed as a props an not defined fixed here. 
If for some reason you want to use this component outside of posts this wont work. Please add all of the options via props, you can defined this values as fallback if nothing is passed

**Review by iruzevic — CHANGES_REQUESTED** [review]

**Review by iruzevic — APPROVED** [review]

**Review by dingo-d — APPROVED** [review]

**Review by IvanGrginovInf — APPROVED** [review]

**Review by kmaric69 — APPROVED** [review]

**Pull request merged**